### PR TITLE
Issue 16933 using browser fingerprint

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 1]
```diff
@@ -98,10 +99,16 @@ public final Response uploadTempResourceMulti(@Context final HttpServletRequest
                 if (fileName == null || fileName.startsWith(".") || fileName.contains("/.")) {
                     continue;
                 }
-                tempFiles.add(tempApi.createTempFile(fileName, user, uniqueKey, in));
+                tempFiles.add(tempApi.createTempFile(fileName, request, in));
             }
 
             return Response.ok(ImmutableMap.of("tempFiles", tempFiles)).build();
+            /*
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: Remove these lines if not used

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 2]
```diff
@@ -179,12 +196,13 @@ private String resolveFileName(final String desiredName, final URL url) {
   
   
   
-  private File createTempPermissionFile(final File parentFolder, List<String> allowList) {
-
+  private File createTempPermissionFile(final File parentFolder, List<String> incomingAccessingList) {
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 3]
```diff
@@ -216,8 +234,10 @@ private File createTempPermissionFile(final File parentFolder, List<String> allo
 
   }
 
-  private boolean canUseTempFile(List<String> accessingList, final DotTempFile dotTempFile) {
+  private boolean canUseTempFile(List<String> incomingAccessingList, final DotTempFile dotTempFile) {
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 4]
```diff
@@ -216,8 +234,10 @@ private File createTempPermissionFile(final File parentFolder, List<String> allo
...
     final File tempFile = dotTempFile.file;
+    List<String> accessingList = new ArrayList<>(incomingAccessingList);
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 5]
```diff
@@ -263,16 +283,20 @@ private boolean canUseTempFile(List<String> accessingList, final DotTempFile dot
    * @param tempFileId
    * @return
    */
-  public Optional<DotTempFile> getTempFile(final User user, final String uniqueKey, final String tempFileId) {
+  public Optional<DotTempFile> getTempFile(final HttpServletRequest request, final String tempFileId) {
     final String anon = Try.of(() -> APILocator.getUserAPI().getAnonymousUser().getUserId()).getOrElse("anonymous");
-
+    final User user = PortalUtil.getUser(request);
+    final String sessionId = (request!=null && request.getSession(false)!=null) ? request.getSession().getId() : null;
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 6]
```diff
@@ -295,5 +319,39 @@ public boolean accept(final File pathname) {
           ;
     }
   };
+  
+  public String getRequestFingerprint(final HttpServletRequest request) {
+    
+    List<String> uniqList = new ArrayList<String>();
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 7]
```diff
@@ -295,5 +319,39 @@ public boolean accept(final File pathname) {
...
+    uniqList.add("User-Agent:" + request.getHeader("User-Agent"));
+    uniqList.add("Host:" + request.getHeader("Host"));
+    uniqList.add("Accept-Language:" + request.getHeader("Accept-Language"));
+
+    uniqList.add("Accept-Encoding:" + request.getHeader("Accept-Encoding"));
+    uniqList.add("X-Forwarded-For:" + request.getHeader("X-Forwarded-For"));
+    uniqList.add("getRemoteHost:" + request.getRemoteHost());
+    uniqList.add("isSecure:" + String.valueOf(request.isSecure()));
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 8]
```diff
@@ -295,5 +319,39 @@ public boolean accept(final File pathname) {
...
+    final String incomingReferer = (request.getHeader("Origin")!=null) ? request.getHeader("Origin") :  request.getHeader("referer");
+    uniqList.add("incomingReferer:" + new SecurityUtils().hostFromUrl(incomingReferer));
+
+    uniqList.add("userId:" + PortalUtil.getUserId(request));
+
+    if(request.getSession(false)!=null) {
+      uniqList.add("getSession:" + request.getSession().getId());
+    }
+    uniqList.removeIf(Objects::isNull);
+    if(uniqList.size()==0) {
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 9]
```diff
@@ -15,7 +15,9 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.NewCookie;
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 10]
```diff
@@ -15,7 +15,9 @@
...
 import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
```
[the remark this answers is not in the document]
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: done

**Review by nollymar — APPROVED** [review]

**Pull request merged**